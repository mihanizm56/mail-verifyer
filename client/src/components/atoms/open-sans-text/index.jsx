import React, { createRef } from "react";
import "./OpenSansText.css";
import sn from "classnames";
import { getClass } from "../../../utils";

export const OpenSansText = ({ text, bold, isButton }: OpenSansTextProps) => (
	<span
		className={sn("open-sans-text", {
			"open-sans-text--button": isButton,
		})}
	>
		{text}
	</span>
);
