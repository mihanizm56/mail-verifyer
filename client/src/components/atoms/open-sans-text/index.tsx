import React from "react";
import classNames from "classnames";
import "./OpenSansText.css";

type OpenSansTextProps = {
	text: string,
	bold?: boolean,
	isButton?: boolean,
};

export const OpenSansText = ({ text, bold, isButton }: OpenSansTextProps) => (
	<span
		className={classNames("open-sans-text", {
			"open-sans-text--button": isButton,
		})}
	>
		{text}
	</span>
);
