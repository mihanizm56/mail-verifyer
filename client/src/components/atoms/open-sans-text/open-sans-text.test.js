import React, { createElement } from "react";
import { shallow } from "enzyme";
import { OpenSansText } from "../open-sans-text";

describe("test OpenSansText", () => {
	beforeEach(() => {
		jest.resetModules();
		jest.clearAllMocks();
	});

	describe("test render contract", () => {
		it("render snap with all props if is button", () => {
			const props = {
				text: "test_text",
				bold: true,
				isButton: true,
			};
			const component = shallow(<OpenSansText {...props} />);

			expect(component).toMatchSnapshot();
		});
		it("render snap with all props if isnt a button and not bold", () => {
			const props = {
				text: "test_text_123",
				bold: false,
				isButton: false,
			};
			const component = shallow(<OpenSansText {...props} />);

			expect(component).toMatchSnapshot();
		});
	});
	describe("test props contract", () => {
		it("render text value", () => {
			const props = {
				text: "test_text_exact",
				bold: false,
				isButton: false,
			};
			const component = shallow(<OpenSansText {...props} />);
			const spanWithText = component.find(".open-sans-text");
			const textValue = spanWithText.props().children;

			expect(textValue).toEqual(props.text);
		});
	});
});
