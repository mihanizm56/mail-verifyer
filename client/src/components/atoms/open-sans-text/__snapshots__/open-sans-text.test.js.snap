// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test OpenSansText test render contract render snap with all props if is button 1`] = `
<span
  className="open-sans-text open-sans-text--button"
>
  test_text
</span>
`;

exports[`test OpenSansText test render contract render snap with all props if isnt a button and not bold 1`] = `
<span
  className="open-sans-text"
>
  test_text_123
</span>
`;
